sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/Hotspots/DoNotHardcodeCredentials.cs

```diff
@@ -93,5 +95,40 @@ private class AssignmentExpressionBannedWordsFinder : CredentialWordsFinderBase<
                 syntaxNode.Left.IsKnownType(KnownType.System_String, semanticModel) &&
                 syntaxNode.Right.IsKind(SyntaxKind.StringLiteralExpression);
         }
+
+        private class StringLiteralBannedWordsFinder : CredentialWordsFinderBase<LiteralExpressionSyntax>
+        {
+            public StringLiteralBannedWordsFinder(DoNotHardcodeCredentialsBase<SyntaxKind> analyzer) : base(analyzer) { }
+
+            protected override string GetAssignedValue(LiteralExpressionSyntax syntaxNode) =>
+                syntaxNode.GetStringValue();
+
+            protected override string GetVariableName(LiteralExpressionSyntax syntaxNode) =>
+                null;
+
+            protected override bool IsAssignedWithStringLiteral(LiteralExpressionSyntax syntaxNode, SemanticModel semanticModel) =>
+                syntaxNode.IsKind(SyntaxKind.StringLiteralExpression) && !IsHandledByOtherFinder(syntaxNode.GetTopMostContainingMethod(), syntaxNode);
```

Comment:
And this one know that he should not handle what the others were doing to avoid collisions.
What I don't like is that the evaluation is done inside function called `IsAssignedWithStringLiteral`, but I didn't want to add another method just for this case since this one is called just where it's needed.

Reply:
> And this one know that he should not handle what the others were doing to avoid collisions.

This is exactly the problem I see. It shouldn't be it's responsibility to know what others are doing. They should be able to tell.